actionpack/lib/action_controller/metal/head.rb

```diff
@@ -24,7 +24,7 @@ def head(status, options = {})
       status ||= :ok
 
       location = options.delete(:location)
-      content_type = options.delete(:content_type)
+      content_type = options.delete(:content_type) || options.delete("Content-Type")
```

Comment:
Let's use the implementation in #28858, `self.content_type ||= …`, with these test cases, or vice versa (add these tests to #28858). That respects the already-set Content-Type header without introducing code that suggests there are two ways of specifying this option.